pkg/services/store/service.go

```diff
@@ -122,71 +124,78 @@ func isFileTypeValid(filetype string) bool {
 	return false
 }
 
-func (s *standardStorageService) Upload(ctx context.Context, user *models.SignedInUser, form *multipart.Form) (*Response, error) {
-	response := Response{
-		path: "upload",
-	}
-	upload, _ := s.tree.getRoot(getOrgId(user), "upload")
+type UploadRequest struct {
+	Contents           []byte
+	MimeType           string
+	Path               string
+	CacheControl       string
+	ContentDisposition string
+	Properties         map[string]string
+
+	OverwriteExistingFile bool
+}
+
+func (s *standardStorageService) Upload(ctx context.Context, user *models.SignedInUser, req UploadRequest) error {
+	upload, _ := s.tree.getRoot(getOrgId(user), RootUpload)
 	if upload == nil {
-		response.statusCode = 404
-		response.message = "upload feature is not enabled"
-		response.err = true
-		return &response, fmt.Errorf("upload feature is not enabled")
+		return ErrUploadFeatureDisabled
 	}
 
-	files := form.File["file"]
-	for _, fileHeader := range files {
-		// Restrict the size of each uploaded file to 1MB based on the header
-		if fileHeader.Size > MAX_UPLOAD_SIZE {
-			response.statusCode = 400
-			response.message = "The uploaded image is too big"
-			response.err = true
-			return &response, nil
-		}
-		// restrict file size based on file size
-		// open each file to copy contents
-		file, err := fileHeader.Open()
-		if err != nil {
-			return nil, err
-		}
-		err = file.Close()
-		if err != nil {
-			return nil, err
-		}
-		data, err := ioutil.ReadAll(file)
+	if !strings.HasPrefix(req.Path, RootUpload+"/") {
+		return ErrUnsupportedFolder
+	}
+
+	validFileType := isFileTypeValid(req.MimeType)
+	if !validFileType {
+		return ErrInvalidFileType
+	}
+
+	if (len(req.Contents)) > MAX_UPLOAD_SIZE {
```

Comment:
For custom branding we'll need to upload files larger than 1mb. Would it be possible to make it configurable?